visualization/dashboard/src/MLIDashboard/Controls/FilterControl.tsx

```diff
@@ -19,87 +21,118 @@ export interface IFilterControlState {
+            borderRadius: '3px',
             display: 'inline-flex',
-            marginRight: "4px"
+            flexDirection:"row",
+            height:"25px"
         },
         filterLabel: {
-            padding: "3px 8px 2px 4px",
-            minWidth: "100px",
-            cursor: "pointer"
+            padding: "1px 9px 6px 11px",
+            minWidth:"75px",
+            maxWidth: "90px",
+            color: "#0078D4",
+            height:"25px",
+            whiteSpace: "nowrap",
+            overflow: "hidden",
+            textOverflow: "ellipsis"
         },
-        filterList: {
-            display: "inline-flex",
-            flexDirection: "row"
+        defaultFilterList: {
+            color: "#979797",
+            marginLeft:"10px"
+        },
+        wrapper: {
+            display: "flex",
+            flexDirection:"column"
+        },
+        addedFilter:
+        {
+            fontWeight: FontWeights.semibold,
+            color: "#000000",
+            marginLeft:"45px",
+            height:"30px",
+            width:"178px"
+        },
+        addedFilterDiv:{
+            marginRight:"40px",
+            marginLeft:"45px",
+            marginBottom:"18px",
+            maxHeight:"97px",
+            overflowY:"auto"
+        },
+        filterIcon:{
+            height:"25px",
+            width:"25px",
         }
+        
     });
     constructor(props: IFilterControlProps) {
         super(props);
-        this.openFilter = this.openFilter.bind(this);
-        this.editNewFilter = this.editNewFilter.bind(this);
+        this.editFilter = this.editFilter.bind(this);
         this.updateFilter = this.updateFilter.bind(this);
         this.cancelFilter = this.cancelFilter.bind(this);
-        this.state = {openedFilter: undefined};
+        this.state = {openedFilter: undefined, filterIndex: this.props.filterContext.filters.length};
     }
 
-    private readonly initialFilter: IFilter = {
-        column: JointDataset.IndexLabel,
-        method: this.props.jointDataset.metaDict[JointDataset.IndexLabel].treatAsCategorical ?
-            FilterMethods.includes : FilterMethods.greaterThan,
-        arg: this.props.jointDataset.metaDict[JointDataset.IndexLabel].treatAsCategorical ?
-            [] : 0
-    };
-
     public render(): React.ReactNode {
         const filterList = this.props.filterContext.filters.map((filter, index) => {
-            return (<div className={FilterControl.classNames.existingFilter}>
-                <div
-                    className={FilterControl.classNames.filterLabel}
-                    onClick={this.openFilter.bind(this, index)}>{this.props.jointDataset.metaDict[filter.column].abbridgedLabel}
-                </div>
+            return (<div key={index} className={FilterControl.classNames.existingFilter}>
+                    <Text variant={"small"} className={FilterControl.classNames.filterLabel}>{this.props.jointDataset.metaDict[filter.column].abbridgedLabel 
```

Comment:
I think we'll want this in a function, to handle cases when the text is too long, like say "Education includes College and 4 others" or "Education includes 4 categories". Having it be a separate function should make it easier to change and test.

Reply:
Also, arg is an array of values for categorical fitlers, we want to check for that. And with the range filter (iin the future) it will be an array of min and max, so will need to be unpacked into a nice form

Reply:
Yeah, I'd suggest a function that takes in a IFilter and returns a React.ReactNode HTML object, so then you just use your map over filters, pass in to that function, and it returns the list of filter HTML objects